### ISSUE TITLE
Temporal reasoning in continuous latent space (Coconut-style)

### ISSUE THREAD

**Issue by justinshenk** [justinshenk/temporal-awareness#10]
## Background

[Coconut (Hao et al., 2024)](https://arxiv.org/abs/2412.06769) shows that LLMs can reason in continuous latent space rather than text, enabling breadth-first exploration of alternatives before committing to output.

**Key insight**: Temporal reasoning is exactly the kind of task that benefits from this - weighing multiple time horizons simultaneously before committing to advice.

## Core Hypothesis

Language constrains temporal reasoning to discrete categories ("short-term", "long-term"). Continuous latent space could represent:
- Continuous temporal gradients
- Probability distributions over horizons
- Simultaneous consideration of multiple futures

### 1. Do continuous thoughts encode richer temporal representations?

**Experiment**: Apply temporal probes to Coconut's latent reasoning steps

```
Standard model: prompt → hidden states → output
                          ↑
                    probe here (single snapshot)

Coconut: prompt → thought₁ → thought₂ → ... → output
                     ↑          ↑
               probe each continuous thought
```

**Hypothesis**: Different thinking steps explore different horizons before converging

### 2. Does latent reasoning enable temporal breadth-first search?

For a decision with short-term vs long-term tradeoffs:
- Does Coconut explore both before committing?
- Can we see "branching" in temporal probe predictions across thoughts?

```
thought₁: [0.5 short, 0.5 long]  # Undecided
thought₂: [0.3 short, 0.7 long]  # Leaning long
thought₃: [0.2 short, 0.8 long]  # Converging
output: "Consider the long-term implications..."
```

### 3. Does latent space enable better temporal calibration?

From issue #6 (uncertainty scaling): Models should hedge more for distant futures.

**Hypothesis**: Coconut shows better temporal calibration because continuous representation can encode uncertainty gradients, not just discrete hedging words.

**Experiment**: Compare temporal uncertainty calibration:
- Standard CoT: Must verbalize uncertainty
- Coconut: Can encode uncertainty in latent space

### 4. Can we detect temporal planning in continuous thoughts?

From issue #5 (eval awareness): Want to detect long-term planning that isn't verbalized.

**With Coconut, this is natural**: Probe the continuous thoughts, not just the output.

**Experiment**:
1. Give Coconut a decision task
2. Monitor continuous thoughts with temporal probes
3. Compare latent temporal content to verbalized output
4. Flag cases where latent reasoning is more long-term than stated

### 5. Can we steer temporal focus in latent space?

**Experiment**:
1. Find temporal direction in continuous thought space
2. Steer toward short-term or long-term during latent reasoning
3. Observe effect on:
   - Final output
   - Quality of reasoning
   - Calibration

### Phase 1: Probe Coconut (2-3 days)

1. Get Coconut implementation (from paper or reimplement)
2. Run on temporal decision tasks
3. Extract continuous thoughts at each step
4. Apply existing Layer 10 probe to each thought
5. Visualize temporal evolution across thinking steps

### Phase 2: Analyze Representations (2-3 days)

1. Do thoughts cluster by temporal horizon?
2. Is there a "temporal convergence" pattern?
3. Compare to standard CoT hidden states

### Phase 3: Temporal-Specific Training (1 week)

1. Create temporal reasoning training set
2. Fine-tune Coconut on temporal decisions
3. Evaluate:
   - Temporal calibration (issue #6)
   - Consistency (issue #7)
   - Register robustness (issue #9)

## Expected Benefits

| Standard Reasoning | Coconut Temporal |
|-------------------|------------------|
| Discrete horizon categories | Continuous temporal gradients |
| Sequential horizon consideration | Parallel exploration |
| Language-constrained uncertainty | Latent uncertainty encoding |
| One reasoning path | Breadth-first temporal search |

## Connection to Other Issues

- **#3 Explicit temporal context**: Coconut could output temporal distributions from latent encoding
- **#5 Eval awareness**: Monitor latent thoughts for hidden long-term planning
- **#6 Uncertainty scaling**: Latent space may encode calibrated uncertainty better
- **#7 Consistency**: Latent reasoning might be more consistent than verbalized
- **#8 Value alignment**: Explore reflective vs immediate in latent space

## Resources

- [Coconut paper](https://arxiv.org/abs/2412.06769)
- [Code (if available)](https://github.com/) - TBD
- Existing temporal probes from probe validation (426f413)

## Success Criteria

1. **Visualization**: Show temporal probe predictions evolving across continuous thoughts
2. **Quantitative**: Coconut shows better temporal calibration than CoT on matched tasks
3. **Interpretability**: Can identify which continuous thoughts explore which horizons
4. **Steering**: Successfully shift temporal focus via latent intervention
